pkg/config/utils/metadata_as_tags.go

```diff
@@ -29,52 +29,73 @@ import (
 // In case of conflict, higher priority configuration value takes precedences
 // For example, if kubernetes_pod_labels_as_tags = {`l1`: `v1`, `l2`: `v2`} and kubernetes_resources_labels_as_tags = {`pods`: {`l1`: `x`}},
 // the resulting labels as tags for pods will be {`l1`: `x`, `l2`: `v2`}
-type MetadataAsTags struct {
+type MetadataAsTags interface {
```

Comment:
Defined an interface to make it easy to create mock implementations in unit tests